host/bcm283x/gpio.go

```diff
@@ -478,10 +506,28 @@ func (p *Pin) WaitForEdge(timeout time.Duration) bool {
 
 // Pull implements gpio.PinIn.
 //
-// bcm283x doesn't support querying the pull resistor of any GPIO pin.
+// bcm2711/bcm2838 support querying the pull resistor of all GPIO pins.
+// Prior to it, bcm283x doesn't support querying the pull resistor of any GPIO pin.
 func (p *Pin) Pull() gpio.Pull {
-	// TODO(maruel): The best that could be added is to cache the last set value
-	// and return it.
+	// sysfs does not have the capability to read pull resistor.
+	if drvGPIO.gpioMemory != nil {
+		if drvGPIO.useLegacyPull {
+			// TODO(maruel): The best that could be added is to cache the last set value
+			// and return it.
+			return gpio.PullNoChange
+		} else {
```

Comment:
This would trigger golint, as in general else after a return is not idiomatic. I can fix in a follow up.

Reply:
Fixed in a follow up.